src/OpenTelemetry.Instrumentation.Dependencies/Implementation/SqlEventSourceListener.netfx.cs

```diff
@@ -0,0 +1,190 @@
+        internal const string ActivityName = ActivitySourceName + ".Execute";
+        internal const string AdoNetEventSourceName = "Microsoft-AdoNet-SystemData";
+        internal const int BeginExecuteEventId = 1;
+        internal const int EndExecuteEventId = 2;
+
+        private static readonly Version Version = typeof(SqlEventSourceListener).Assembly.GetName().Version;
+        private static readonly ActivitySource SqlClientActivitySource = new ActivitySource(ActivitySourceName, Version.ToString());
+
+        private readonly SqlClientInstrumentationOptions options;
+        private EventSource eventSource;
+
+        public SqlEventSourceListener(SqlClientInstrumentationOptions options = null)
+        {
+            this.options = options ?? new SqlClientInstrumentationOptions();
+        }
+
+        public override void Dispose()
+        {
+            if (this.eventSource != null)
+            {
+                this.DisableEvents(this.eventSource);
+            }
+
+            base.Dispose();
+        }
+
+        protected override void OnEventSourceCreated(EventSource eventSource)
+        {
+            if (eventSource?.Name.StartsWith(AdoNetEventSourceName) == true)
+            {
+                this.eventSource = eventSource;
+                this.EnableEvents(eventSource, EventLevel.Informational, (EventKeywords)1);
+            }
+
+            base.OnEventSourceCreated(eventSource);
+        }
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            try
+            {
+                if (eventData.EventId == BeginExecuteEventId)
+                {
+                    this.OnBeginExecute(eventData);
+                }
+                else if (eventData.EventId == EndExecuteEventId)
+                {
+                    this.OnEndExecute(eventData);
+                }
+            }
+            catch (Exception exc)
+            {
+                InstrumentationEventSource.Log.UnknownErrorProcessingEvent(nameof(SqlEventSourceListener), nameof(this.OnEventWritten), exc);
+            }
+        }
+
+        private void OnBeginExecute(EventWrittenEventArgs eventData)
+        {
+            /*
+               Expected payload:
+                [0] -> ObjectId
+                [1] -> DataSource
+                [2] -> Database
+                [3] -> CommandText ([3] = CommandType == CommandType.StoredProcedure ? CommandText : string.Empty)
+             */
+
+            if ((eventData?.Payload?.Count ?? 0) < 4)
+            {
+                InstrumentationEventSource.Log.InvalidPayload(nameof(SqlEventSourceListener), nameof(this.OnBeginExecute));
+                return;
+            }
+
+            var activity = SqlClientActivitySource.StartActivity(ActivityName, ActivityKind.Client);
+            if (activity == null)
+            {
+                return;
+            }
+
+            string databaseName = (string)eventData.Payload[2];
```

Comment:
wondering why setting displayname is not inside `if (activity.IsAllDataRequested)` ?

Reply:
I modeled after the HttpWebRequestActivitySource:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/ca322b3883b93b1a8784676549c944a936e3a37d/src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.netfx.cs#L94-L110
Line 96 in that snippet.

I did it there originally because the DisplayName felt useful to have always? Open to change it.

Reply:
Got. I couldn't find any guidelines on this, so at this points, its up to individual library authors to do it. (in case adapters "we" are the authors).